pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -200,160 +149,163 @@ func (c *Controller) updateShootStatusProcessing(shoot *gardenv1beta1.Shoot, mes
+			lastErr := gardencorev1alpha1helper.LastError(fmt.Sprintf("Shoot is not on healthy seed: %v", err))
+			c.recorder.Event(shoot, corev1.EventTypeWarning, gardenv1beta1.EventDeleteError, lastErr.Description)
+			if err := c.updateShootStatusDeleteError(o, lastErr); err != nil {
+				o.Logger.WithError(err).Error("Could not update the Shoot status after deletion error")
+				return reconcile.Result{}, err
 			}
-			if syncPeriodDuration >= time.Minute {
-				syncPeriod = metav1.Duration{Duration: syncPeriodDuration}
+			return reconcile.Result{}, err
+		}
+
+		if err := o.SyncClusterResourceToSeed(context.TODO()); err != nil {
+			lastErr := gardencorev1alpha1helper.LastError(fmt.Sprintf("Could not sync the Cluster resource to the Seed: %v", err))
+			o.Logger.Error(lastErr.Description)
+			if err := c.updateShootStatusDeleteError(o, lastErr); err != nil {
+				o.Logger.WithError(err).Error("Could not update the Shoot status after deletion error")
+				return reconcile.Result{}, err
 			}
+			return reconcile.Result{}, err
 		}
 	}
 
-	var (
-		syncPeriodNano = syncPeriod.Nanoseconds()
-		nextSyncNano   = currentTimeNano - (currentTimeNano-creationTimeNano)%syncPeriodNano + syncPeriodNano
-	)
-
-	return time.Duration(nextSyncNano - currentTimeNano)
-}
-
-// ControlInterface implements the control logic for updating Shoots. It is implemented as an interface to allow
-// for extensions that provide different semantics. Currently, there is only one implementation.
-type ControlInterface interface {
-	// ReconcileShoot implements the control logic for Shoot creation, update, and deletion.
-	// If an implementation returns a non-nil error, the invocation will be retried using a rate-limited strategy.
-	// Implementors should sink any errors that they do not wish to trigger a retry, and they may feel free to
-	// exit exceptionally at any point provided they wish the update to be re-run at a later point in time.
-	// The bool return value determines whether the Shoot should be automatically requeued for reconciliation.
-	ReconcileShoot(shoot *gardenv1beta1.Shoot, key string, o *operation.Operation) (bool, error)
-}
-
-// NewDefaultControl returns a new instance of the default implementation ControlInterface that
-// implements the documented semantics for Shoots. updater is the UpdaterInterface used
-// to update the status of Shoots. You should use an instance returned from NewDefaultControl() for any
-// scenario other than testing.
-func NewDefaultControl(k8sGardenClient kubernetes.Interface, k8sGardenInformers gardeninformers.Interface, secrets map[string]*corev1.Secret, imageVector imagevector.ImageVector, identity *gardenv1beta1.Gardener, config *config.ControllerManagerConfiguration, gardenerNamespace string, recorder record.EventRecorder) ControlInterface {
-	return &defaultControl{k8sGardenClient, k8sGardenInformers, secrets, imageVector, identity, config, gardenerNamespace, recorder}
-}
+	if common.IsShootFailed(shoot) {
```

Comment:
I always wanted to 'greedily' sync the cluster resource to the extensions so they also see if a Shoot is ignored or not - otherwise, this would be lost for them / any other updates in between wouldn't reach them.